support-frontend/assets/pages/contributions-checkout-in-epic/ContributionsCheckout.tsx

```diff
@@ -0,0 +1,166 @@
+import { css } from '@emotion/react';
+import { ResizeObserver } from '@juggle/resize-observer';
+import React, { useEffect } from 'react';
+import useDimensions from 'react-cool-dimensions';
+import type { ConnectedProps } from 'react-redux';
+import { connect } from 'react-redux';
+import type { ThunkDispatch } from 'redux-thunk';
+import type {
+	ContributionAmounts,
+	ContributionType,
+} from 'helpers/contributions';
+import { getAmount } from 'helpers/contributions';
+import type { Action } from 'pages/contributions-landing/contributionsLandingActions';
+import {
+	selectAmount,
+	updateContributionTypeAndPaymentMethod,
+	updateOtherAmount,
+} from 'pages/contributions-landing/contributionsLandingActions';
+import type { State } from 'pages/contributions-landing/contributionsLandingReducer';
+import { ContributionsCheckoutForm } from './ContributionsCheckoutForm';
+import { ContributionsCheckoutSubmitting } from './ContributionsCheckoutSubmitting';
+import { ContributionsCheckoutThankYou } from './ContributionsCheckoutThankYou';
+
+// ---- Component ---- //
+
+type Status = 'FORM' | 'SUBMITTING' | 'SUCCESS';
```

Comment:
Perhaps this could be better named - `FORM` really means `The user is interacting with the form`. But I was struggling to think of something nice and succinct!

Reply:
Possibly something like `INPUT` or `ENTRY`? I don't think `FORM` is a huge problem though.